core/station/impl/src/services/external_canister.rs

```diff
@@ -718,32 +770,176 @@ impl ExternalCanisterService {
+                            Resource::ExternalCanister(ExternalCanisterResourceAction::Call(
+                                CallExternalCanisterResourceTarget {
+                                    execution_method:
+                                        ExecutionMethodResourceTarget::ExecutionMethod(
+                                            CanisterMethod {
+                                                canister_id: external_canister.canister_id,
+                                                method_name: call.execution_method,
+                                            },
+                                        ),
+                                    validation_method: call.validation_method,
+                                },
+                            ));
+
+                        self.permission_service
+                            .edit_permission(EditPermissionOperationInput {
+                                auth_scope: Some(call.allow.auth_scope),
+                                users: Some(call.allow.users),
+                                user_groups: Some(call.allow.user_groups),
+                                resource: call_resource.clone(),
+                            })?;
+                    }
+                }
+                ExternalCanisterChangeCallPermissionsInput::OverrideSpecifiedByExecutionMethods(
+                    calls,
+                ) => {
+                    let update_call_methods = calls.iter().fold(HashSet::new(), |mut acc, call| {
+                        acc.insert(call.execution_method.clone());
+                        acc
+                    });
+
+                    // removes all existing call permissions of the updated methods
+                    self.permission_repository
+                        .find_external_canister_call_permissions(&external_canister.canister_id)
+                        .iter()
+                        .filter(|permission| {
+                            if let Resource::ExternalCanister(
+                                ExternalCanisterResourceAction::Call(
+                                    CallExternalCanisterResourceTarget {
+                                        execution_method:
+                                            ExecutionMethodResourceTarget::ExecutionMethod(
+                                                CanisterMethod {
+                                                    canister_id: _,
+                                                    method_name,
+                                                },
+                                            ),
+                                        validation_method: _,
+                                    },
+                                ),
+                            ) = &permission.resource
+                            {
+                                update_call_methods.contains(method_name)
+                            } else {
+                                false
```

Comment:
we should never reach this branch, right?

Reply:
correct, i've added a `matches!` here to be more concise. throwing an error would also be an option but maybe we could simply ignore it and not block the execution

Reply:
sure, let's leave it with the matches!